api/logical.go

```diff
@@ -82,8 +82,8 @@ func (c *Logical) List(path string) (*Secret, error) {
 		defer resp.Body.Close()
 	}
 	if resp != nil && resp.StatusCode == 404 {
-		secret, err := ParseSecret(resp.Body)
-		switch err {
+		secret, parseErr := ParseSecret(resp.Body)
+		switch parseErr {
```

Comment:
Can we wrap this error with `err`?

Reply:
We swallow the error normally in this case, so we don't really need to wrap it -- otherwise there wouldn't be any error returned. See the updated code.